tools/ci/src/main.rs

```diff
@@ -46,7 +46,7 @@ fn main() {
     if what_to_run.contains(Check::CLIPPY) {
         // See if clippy has any complaints.
         // - Type complexity must be ignored because we use huge templates for queries
-        cmd!("cargo clippy --workspace --all-targets --all-features -- -A clippy::type_complexity -W clippy::doc_markdown -D warnings")
+        cmd!("cargo clippy --workspace --all-targets --all-features -- -A clippy::type_complexity -W clippy::doc_markdown -W clippy::option_if_let_else -W clippy::redundant_else -W clippy::match_same_arms -W clippy::semicolon_if_nothing_returned -W clippy::explicit_iter_loop -W clippy::map_flatten -D warnings")
```

Comment:
Can we use some string parsing to list out these lints in a cleaner fashion?

Reply:
I tried many different ways but `xshell` (even updated to 0.2) keeps printing the flags between double quotes, so I gave up

Reply:
I'm pretty sure that `xshell` doesn't do argument splitting outside `cmd!` macro, so we need to keep the flags in a vec and expand like `cmd!("cargo clippy {args...}")`

Reply:
I tried that, and it still printed every flag between double quotes even though I'm sure I did exactly as the `xshell` doc does it

Reply:
@infmagic2047 wether `flags` are stored in a `Vec<&str>`, a `Vec<String>`, or a `[&str]`, `const` or  in a `let`:
```rust
let flags = vec![
            "-A clippy::type_complexity",
            "-W clippy::doc_markdown",
            "-W clippy::option_if_let_else",
            "-W clippy::redundant_else",
            "-W clippy::match_same_arms",
            "-W clippy::semicolon_if_nothing_returned",
            "-W clippy::explicit_iter_loop",
            "-W clippy::map_flatten",
            "-D warnings",
        ];
cmd!("cargo clippy --workspace --all-targets --all-features -- {flags...}")
```

This is the output:

`cargo clippy --workspace --all-targets --all-features -- "-A clippy::type_complexity" "-W clippy::doc_markdown" "-W clippy::option_if_let_else" "-W clippy::redundant_else" "-W clippy::match_same_arms" "-W clippy::semicolon_if_nothing_returned" "-W clippy::explicit_iter_loop" "-W clippy::map_flatten" "-D warnings"`

Reply:
`-A clippy::type_complexity` is actually two arguments `-A` and `clippy::type_complexity`, so writing them as one won't work. You can either write them as two elements, or use the concatenated form `-Aclippy::type_complexity`.

Reply:
Thanks that seems to work, committing now